features/step_definitions/auth_steps.rb

```diff
@@ -0,0 +1,32 @@
+Given(/^I have an account on the site$/) do
+  @user = User.last
```

Comment:
There shouldn't be any `User`s in the DB when a test begins to run.

If you have leftover users from your previous tests, you should use DatabaseCleaner to wipe them out.

Once you do that, you should use a factory here to create the user you need.